polynote-kernel/src/main/scala/polynote/kernel/util/LimitedSharingClassLoader.scala

```diff
@@ -27,9 +27,12 @@ class LimitedSharingClassLoader(
   override def loadClass(name: String, resolve: Boolean): Class[_] =  getClassLoadingLock(name).synchronized {
     val c = if (share.test(name)) {
       //System.err.println(s"Delegating class $name")
-      super.loadClass(name, resolve)
-    }
-    else try {
+      try {
+        super.loadClass(name, resolve)
+      } catch {
+        case err: ClassNotFoundException => findClass(name)
```

Comment:
What is this addition for? Were we just breaking the classloader contract by not doing this?

Reply:
Well, there might be cases where you do have classes from one of the "blessed" packages that are in your dependencies rather than Spark's. So we should exhaust that possibility before giving up.